js/ui/codingManager.js

```diff
@@ -927,11 +1035,174 @@ const CodingSession = new Lang.Class({
     }
 });
 
-const CodingManager = new Lang.Class({
-    Name: 'CodingManager',
+// Given a FlatpakInstalledRef app, look up its exports directory
+// and find any relevant .desktop files and add them to the desktop
+// if they are not already added.
+function addFlatpakAppToDesktop(app) {
+    // First, get the exports directory. We'll look in here to find
+    // any exported .desktop files
+    let deploy = app.get_deploy_dir();
+    let applications = Gio.File.new_for_path(GLib.build_filenamev([
+        deploy,
+        'export',
+        'share',
+        'applications'
+    ]));
+
+    let enumerator = applications.enumerate_children('standard::*',
+                                                     Gio.FileQueryInfoFlags.NONE,
+                                                     null);
+
+    let childInfo = null;
+    while ((childInfo = enumerator.next_file(null)) !== null) {
+        let name = childInfo.get_name();
+        if (name.endsWith('.desktop')) {
+            if (!IconGridLayout.layout.iconIsFolder(name)) {
+                IconGridLayout.layout.appendIcon(name,
+                                                 IconGridLayout.DESKTOP_GRID_ID);
+            }
+        }
+    }
+}
+
+const OS_INTEGRATION_APP = 'org.gnome.Weather.Application';
+
+const CodingOSIntegration = new Lang.Class({
+    Name: 'CodingOSIntegration',
+
+    _init: function(manager) {
+        let listeningFor = Set();
+
+        manager.connect('session-created', Lang.bind(this, function(instance, session) {
+            // Check to see if the session was created on org.gnome.Weather.Application
+            if (session.app.meta_window.get_flatpak_id() !== OS_INTEGRATION_APP)
```

Comment:
I don't understand this check here; my understanding is that if we got here, we are already on a session for one of the whitelisted apps.